src/Symfony/Component/Routing/Tests/Generator/UrlGeneratorTest.php

```diff
@@ -128,6 +131,32 @@ public function testUrlWithNullExtraParameters()
         $this->assertEquals('http://localhost/app.php/testing', $url);
     }
 
+    public function testRelativeUrlWithUidParameters()
+    {
+        $ulid = new Ulid();
+        $uuid = new UuidV4();
+
+        $routes = $this->getRoutes('test', new Route('/testing'));
+        $url = $this->getGenerator($routes)->generate('test', ['ulid' => $ulid, 'uuid' => $uuid], UrlGeneratorInterface::ABSOLUTE_PATH);
+
+        $expected = sprintf('/app.php/testing?ulid=%s&uuid=%s', $ulid->__toString(), $uuid->__toString());
```

Comment:
there's no need to call explicitly __toString. that's the point of magic methods :)

Reply:
I know I just wanted to make sure the test works as expected.